### PR TITLE
dockerfile and dockerignore

### PR DESCRIPTION
**Pull request by Fenyo7**

### CODE REVIEW

**Review by Jankita58 — APPROVED** [review]

File: sk0-back/dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+RUN npm install
+
+COPY . .
+
+RUN npm run build
+
+# Stage 2: Run
+FROM node:20-alpine
+
+ENV PORT=3000
```
Comment (rolcho) [inline sk0-back/dockerfile]: Where checks the app the environment variable? I can't see in the main.ts file it seems hard-coded:
```typescript
import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  await app.listen(3000);
}
bootstrap();

File: sk0-back/dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Stage 1: Build for local development
```
Comment (rolcho) [inline sk0-back/dockerfile]: As a convention the filename starts with capital 'D' -> Dockerfile

File: sk0-back/src/main.ts [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
 async function bootstrap() {
+  const env = process.env.NODE_ENV || 'development';
+  const envConfig = dotenv.parse(fs.readFileSync(`.env.${env}`));
+  for (const k in envConfig) {
+    process.env[k] = envConfig[k];
+  }
   const app = await NestFactory.create(AppModule);
-  await app.listen(3000);
+  const port = process.env.PORT;
+  await app.listen(port);
```
Comment (rolcho) [inline sk0-back/src/main.ts]: I have tried to start in a docker container with the env PORT set to 3005 in the Dockerfile, but didn't start for me

Comment (Fenyo7) [reply sk0-back/src/main.ts]: I sort of fixed this, but when I tried for some reason it started listening using ipv6 addressing, and the page it linked to didn't respond, so I had to hard code to open localhost, which is absolutely horrible, I'll need some help tomorrow with this

**Review by rolcho — APPROVED** [review]
🤘🚀

**Review by markkovari — APPROVED** [review]

**Pull request merged**